action/metadata-support.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/google/go-github/v45/github"
+	"github.com/sethvargo/go-githubactions"
+	"golang.org/x/oauth2"
+)
+
+// getArtifacts queries github for artifact list, returning a map of strings of
+// artifact names paired with their variant type.
+// org, repo, and workflowrunID are the github vars passed in to the API for the
+// workflow run query, see: https://docs.github.com/en/rest/actions/artifacts#list-workflow-run-artifacts
+func getArtifacts(org string, repo string, workflowRunID int64) map[string][]string {
+
+	// Auth to github API.
+	token := os.Getenv("GITHUB_TOKEN")
+	if token == "" {
+		githubactions.Errorf("missing env GITHUB_TOKEN")
+	}
+
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+	ghClient := github.NewClient(tc)
+
+	// Query API for list of artifacts associated with specified build workflow of product.
+	opt := &github.ListOptions{PerPage: 100} // 100 is max resposes per page.
+	var artifacts []*github.ArtifactList     // each page of results is an element in array.
+	for {
+		artifactPage, response, err := ghClient.Actions.ListWorkflowRunArtifacts(ctx, org, repo, workflowRunID, opt)
+		if err != nil {
+			fmt.Println(err)
+		}
+		artifacts = append(artifacts, artifactPage)
+		if response.NextPage == 0 {
+			break
+		}
+		opt.Page = response.NextPage
+	}
+
+	// Extract a list of artifact names from the artifact data structure returned from the
+	// github API.
+	var rawArtifacts []string
+	for i := range artifacts {
+		for j := range artifacts[i].Artifacts {
+			rawArtifacts = append(rawArtifacts, *artifacts[i].Artifacts[j].Name)
+		}
+	}
```

Comment:
The `range` keyword produces two values, the index and the value  (or key/value in the case of iterating over a map). In many cases you only want the value, so you can ignore the index by assigning it to `_`.  A more idiomatic way of writing this would be:
```suggestion
	for _, ghArtifact := range artifacts {
  		for _, artifact := range ghArtifact.Artifacts {
   	 		rawArtifacts = append(rawArtifacts, *artifact.Name)
 	 	}
	}
```

The `*` before `artifact.Name` might not be needed, and should be tested.